### PR TITLE
GFM-4 - Added Typeahead country feature

### PR DESCRIPTION
**Pull request by gxxm**
This adds the typeahead country dropdown.

### CODE REVIEW

**Comment by easternbloc** [comment]
I'd kinda like us to create a module for this list of countries as it's the HO list of countries and will likely be reused ALOT. I'm cool with this but consider that to be your technical debt claxon 😄

**Comment by JoeChapman** [comment]
@gxxm This needs some ES6 love, `const`, `let`, arrow functions etc...

File: assets/js/index.js [hunk 1]
```diff
@@ -7,3 +7,72 @@ var formFocus = toolkit.formFocus;
...
+  var $input = $('<input/>');
+  var selectedValue = $el.val();
+  var typeaheadList = $el.find('option').map(function mapOptions() {
+    if (this.value === '') {
+      // remove any empty values from typeahead
+      /*eslint consistent-return: 0*/
+      return;
+      /*eslint consistent-return: 1*/
+    }
+    return this.value;
```
Comment (JoeChapman) [inline assets/js/index.js]:
```
    return this.value === '' ? undefined : this.value;
```

**Comment by JoeChapman** [comment]
I hate jQuery

**Comment by JoeChapman** [comment]
@gxxm what are you doing with this pull request? Please can you address the issues in this pull request?

**Pull request closed without merging**